frontend/src/components/FundsPullPreApproval/ApproveModal.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import { Button, Modal, ModalBody } from "reactstrap";
+import CloseButton from "../CloseButton";
+import { Approval } from "../../interfaces/approval";
+import ApprovalDetails from "./ApprovalDetails";
+import BackendClient from "../../services/backendClient";
+
+interface ApproveModalProps {
+  approval: Approval | undefined;
+  open: boolean;
+  onClose: () => void;
+}
+
+function ApproveModal({ approval, open, onClose }: ApproveModalProps) {
+  const updateApproval = async () => {
+    try {
+      await new BackendClient().updateApprovalStatus(approval!.funds_pull_pre_approval_id, "valid");
```

Comment:
there is no indication for what's going on. The user will just hand until the request is complete and then nothing will happen, the user won't know that it succeeded / failed / still going... Manage the request state, add a spinner or some indicator request is still going, display errors if any. Finally when the request is complete display a success message and close the modal.